pandas/core/series.py

```diff
@@ -1250,6 +1250,10 @@ def setitem(key, value):
     def _set_with_engine(self, key, value):
         values = self._values
         try:
+            if is_extension_array_dtype(values):
```

Comment:
Is `values` an array-like or list-like? Can we do `is_extension_array_dtype(values.dtype)`?

Reply:
Yup. Changed to pass the dtype instead